modules/tensor_mechanics/src/materials/ADLAROMCreepStressUpdate.C

```diff
@@ -0,0 +1,535 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADLAROMCreepStressUpdate.h"
+
+#include "Function.h"
+#include "MathUtils.h"
+
+registerADMooseObject("TensorMechanicsApp", ADLAROMCreepStressUpdate);
+
+defineADValidParams(
+    ADLAROMCreepStressUpdate,
+    ADRadialReturnCreepStressUpdateBase,
+    params.addClassDescription(
+        "Calculates the effective creep strain based on the rates predicted by a material "
+        "specific Los Alamos Reduced Order Model derived from a Visco-Plastic Self Consistent "
+        "calculations.");
+
+    params.addRequiredParam<UserObjectName>("rom_data", "Name of user object holding ROM data");
+    params.addRequiredCoupledVar("temperature", "The coupled temperature (K)");
+    params.addCoupledVar("environmental_factor", 0.0, "Optional coupled environmental factor");
+    params.addRangeCheckedParam<Real>("input_window_limit",
+                                      1.0,
+                                      "input_window_limit>0.0",
+                                      "Multiplier for the input minium/maximum input window");
+    MooseEnum window_failure("ERROR WARN IGNORE", "WARN");
+    params.addParam<MooseEnum>("input_window_failure_action",
+                               window_failure,
+                               "What to do if ROM input is outside the window of applicability.");
+    params.addParam<bool>("extrapolate_to_zero_stress",
+                          true,
+                          "Flag to allow for extrapolation of input J2 stress to zero");
+
+    params.addRangeCheckedParam<Real>("initial_mobile_dislocation_density",
+                                      0.0,
+                                      "initial_mobile_dislocation_density >= 0.0",
+                                      "Initial density of mobile (glissile) dislocations (1/m^2)");
+    params.addParam<Real>(
+        "max_relative_mobile_dislocation_increment",
+        "Maximum increment of density of mobile (glissile) dislocations (1/m^2). If provided, this "
+        "will override the increment provided with the ROM data.");
+    params.addParam<FunctionName>(
+        "mobile_dislocation_density_forcing_function",
+        "Optional forcing function for immobile dislocation. If provided, the immobile dislocation "
+        "density will be reset to the function value at the beginning of the timestep. Used for "
+        "testing purposes only.");
```

Comment:
I agree that the testing is important, but I'm not convinced that this capability belongs in the final code here. Definitely worth further discussion because the testing is an important component

Reply:
If all the rom data is folded into a single ROM material, than these must exist. I can move them to "advanced" or even hidden parameters. This is a very easy and minimally invasive way to allow for testing.